src/Platforms/SQLitePlatform.php

```diff
@@ -917,7 +933,11 @@ private function getForeignKeysInAlteredTable(TableDiff $diff): array
                         ->getValue(),
                 );
 
-                $foreignKeys[$constraintKey] = $constraint;
+                if (isset($keysByName[$constraintKey])) {
+                    $foreignKeys[$keysByName[$constraintKey]] = $constraint;
+                } else {
+                    $foreignKeys[] = $constraint;
```

Comment:
This code doesn't need to be duplicated. The logic, at a high level, should look like the following:
```
if the constraint is named
    make sure that there's no existing constraint with the same name
    update the name-key map
end

add the new constraint
```